A job failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/9991345212/job/27613972364) [(plain)](https://github.com/rust-lang/rust/commit/bfbfb84d4edd26828b5ae6a4281893366fdfc2c1/checks/27613972364/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
# Check library crates on all tier 1 targets.
# We disable optimized compiler built-ins because that requires a C toolchain for the target.
# We also skip the x86_64-unknown-linux-gnu target as it is well-tested by other jobs.
ENV SCRIPT python3 ../x.py check --stage 0 --set build.optimized-compiler-builtins=false core alloc std --target=aarch64-unknown-linux-gnu,i686-pc-windows-msvc,i686-unknown-linux-gnu,x86_64-apple-darwin,x86_64-pc-windows-gnu,x86_64-pc-windows-msvc && \
           # Ensure that default config profiles are not broken
           for config_file in ../src/bootstrap/defaults/*.toml; do python3 ../x.py build --dry-run --config "$config_file"; done
           python3 ../x.py clippy bootstrap -Dwarnings && \
           python3 ../x.py clippy compiler library -Aclippy::all -Dclippy::correctness && \
           python3 ../x.py build --stage 0 src/tools/build-manifest && \
           python3 ../x.py test --stage 0 src/tools/compiletest && \
---
#1 DONE 2.4s

#2 [internal] load build definition from Dockerfile
#2 transferring dockerfile: 3.37kB done
#2 WARN: ConsistentInstructionCasing: Command 'python3' should match the case of the command majority (uppercase) (line 51)

 1 warning found (use --debug to expand):
 1 warning found (use --debug to expand):
 - ConsistentInstructionCasing: Command 'python3' should match the case of the command majority (uppercase) (line 51)
--------------------
--------------------
  50 |                for config_file in ../src/bootstrap/defaults/*.toml; do python3 ../x.py build --dry-run --config "$config_file"; done
  51 | >>>            python3 ../x.py check --target=i686-pc-windows-gnu --host=i686-pc-windows-gnu && \
  52 | >>>            python3 ../x.py clippy bootstrap -Dwarnings && \
  53 | >>>            python3 ../x.py clippy compiler library -Aclippy::all -Dclippy::correctness && \
  54 | >>>            python3 ../x.py build --stage 0 src/tools/build-manifest && \
  55 | >>>            python3 ../x.py test --stage 0 src/tools/compiletest && \
  56 | >>>            python3 ../x.py test --stage 0 core alloc std test proc_macro && \
  57 | >>>            # Build both public and internal documentation.
  58 | >>>            RUSTDOCFLAGS=\"--document-private-items --document-hidden-items\" python3 ../x.py doc --stage 0 library && \
  59 | >>>            mkdir -p /checkout/obj/staging/doc && \
  60 | >>>            cp -r build/x86_64-unknown-linux-gnu/doc /checkout/obj/staging && \
  61 | >>>            RUSTDOCFLAGS=\"--document-private-items --document-hidden-items\" python3 ../x.py doc --stage 0 compiler && \
  62 | >>>            RUSTDOCFLAGS=\"--document-private-items --document-hidden-items\" python3 ../x.py doc --stage 0 library/test && \
  63 | >>>            /scripts/validate-toolstate.sh && \
  64 | >>>            /scripts/validate-error-codes.sh && \
  65 | >>>            reuse --include-submodules lint && \
  66 | >>>            # Runs checks to ensure that there are no issues in our JS code.
  67 | >>>            es-check es2019 ../src/librustdoc/html/static/js/*.js && \
  68 | >>>            eslint -c ../src/librustdoc/html/static/.eslintrc.js ../src/librustdoc/html/static/js/*.js && \
  69 | >>>            eslint -c ../src/tools/rustdoc-js/.eslintrc.js ../src/tools/rustdoc-js/tester.js && \
  70 | >>>            eslint -c ../src/tools/rustdoc-gui/.eslintrc.js ../src/tools/rustdoc-gui/tester.js
--------------------
ERROR: failed to solve: dockerfile parse error on line 51: unknown instruction: python3
Command failed. Attempt 2/5:
#0 building with "elastic_dubinsky" instance using docker-container driver
#0 building with "elastic_dubinsky" instance using docker-container driver

#1 [internal] load build definition from Dockerfile
#1 transferring dockerfile: 3.37kB done
#1 WARN: ConsistentInstructionCasing: Command 'python3' should match the case of the command majority (uppercase) (line 51)

 1 warning found (use --debug to expand):
 1 warning found (use --debug to expand):
 - ConsistentInstructionCasing: Command 'python3' should match the case of the command majority (uppercase) (line 51)
--------------------
--------------------
  50 |                for config_file in ../src/bootstrap/defaults/*.toml; do python3 ../x.py build --dry-run --config "$config_file"; done
  51 | >>>            python3 ../x.py check --target=i686-pc-windows-gnu --host=i686-pc-windows-gnu && \
  52 | >>>            python3 ../x.py clippy bootstrap -Dwarnings && \
  53 | >>>            python3 ../x.py clippy compiler library -Aclippy::all -Dclippy::correctness && \
  54 | >>>            python3 ../x.py build --stage 0 src/tools/build-manifest && \
  55 | >>>            python3 ../x.py test --stage 0 src/tools/compiletest && \
  56 | >>>            python3 ../x.py test --stage 0 core alloc std test proc_macro && \
  57 | >>>            # Build both public and internal documentation.
  58 | >>>            RUSTDOCFLAGS=\"--document-private-items --document-hidden-items\" python3 ../x.py doc --stage 0 library && \
  59 | >>>            mkdir -p /checkout/obj/staging/doc && \
  60 | >>>            cp -r build/x86_64-unknown-linux-gnu/doc /checkout/obj/staging && \
  61 | >>>            RUSTDOCFLAGS=\"--document-private-items --document-hidden-items\" python3 ../x.py doc --stage 0 compiler && \
  62 | >>>            RUSTDOCFLAGS=\"--document-private-items --document-hidden-items\" python3 ../x.py doc --stage 0 library/test && \
  63 | >>>            /scripts/validate-toolstate.sh && \
  64 | >>>            /scripts/validate-error-codes.sh && \
  65 | >>>            reuse --include-submodules lint && \
  66 | >>>            # Runs checks to ensure that there are no issues in our JS code.
  67 | >>>            es-check es2019 ../src/librustdoc/html/static/js/*.js && \
  68 | >>>            eslint -c ../src/librustdoc/html/static/.eslintrc.js ../src/librustdoc/html/static/js/*.js && \
  69 | >>>            eslint -c ../src/tools/rustdoc-js/.eslintrc.js ../src/tools/rustdoc-js/tester.js && \
  70 | >>>            eslint -c ../src/tools/rustdoc-gui/.eslintrc.js ../src/tools/rustdoc-gui/tester.js
--------------------
ERROR: failed to solve: dockerfile parse error on line 51: unknown instruction: python3
Command failed. Attempt 3/5:
#0 building with "elastic_dubinsky" instance using docker-container driver
#0 building with "elastic_dubinsky" instance using docker-container driver

#1 [internal] load build definition from Dockerfile
#1 transferring dockerfile: 3.37kB done
#1 WARN: ConsistentInstructionCasing: Command 'python3' should match the case of the command majority (uppercase) (line 51)

 1 warning found (use --debug to expand):
 1 warning found (use --debug to expand):
 - ConsistentInstructionCasing: Command 'python3' should match the case of the command majority (uppercase) (line 51)
--------------------
--------------------
  50 |                for config_file in ../src/bootstrap/defaults/*.toml; do python3 ../x.py build --dry-run --config "$config_file"; done
  51 | >>>            python3 ../x.py check --target=i686-pc-windows-gnu --host=i686-pc-windows-gnu && \
  52 | >>>            python3 ../x.py clippy bootstrap -Dwarnings && \
  53 | >>>            python3 ../x.py clippy compiler library -Aclippy::all -Dclippy::correctness && \
  54 | >>>            python3 ../x.py build --stage 0 src/tools/build-manifest && \
  55 | >>>            python3 ../x.py test --stage 0 src/tools/compiletest && \
  56 | >>>            python3 ../x.py test --stage 0 core alloc std test proc_macro && \
  57 | >>>            # Build both public and internal documentation.
  58 | >>>            RUSTDOCFLAGS=\"--document-private-items --document-hidden-items\" python3 ../x.py doc --stage 0 library && \
  59 | >>>            mkdir -p /checkout/obj/staging/doc && \
  60 | >>>            cp -r build/x86_64-unknown-linux-gnu/doc /checkout/obj/staging && \
  61 | >>>            RUSTDOCFLAGS=\"--document-private-items --document-hidden-items\" python3 ../x.py doc --stage 0 compiler && \
  62 | >>>            RUSTDOCFLAGS=\"--document-private-items --document-hidden-items\" python3 ../x.py doc --stage 0 library/test && \
  63 | >>>            /scripts/validate-toolstate.sh && \
  64 | >>>            /scripts/validate-error-codes.sh && \
  65 | >>>            reuse --include-submodules lint && \
  66 | >>>            # Runs checks to ensure that there are no issues in our JS code.
  67 | >>>            es-check es2019 ../src/librustdoc/html/static/js/*.js && \
  68 | >>>            eslint -c ../src/librustdoc/html/static/.eslintrc.js ../src/librustdoc/html/static/js/*.js && \
  69 | >>>            eslint -c ../src/tools/rustdoc-js/.eslintrc.js ../src/tools/rustdoc-js/tester.js && \
  70 | >>>            eslint -c ../src/tools/rustdoc-gui/.eslintrc.js ../src/tools/rustdoc-gui/tester.js
--------------------
ERROR: failed to solve: dockerfile parse error on line 51: unknown instruction: python3
Command failed. Attempt 4/5:
#0 building with "elastic_dubinsky" instance using docker-container driver
#0 building with "elastic_dubinsky" instance using docker-container driver

#1 [internal] load build definition from Dockerfile
#1 transferring dockerfile: 3.37kB done
#1 WARN: ConsistentInstructionCasing: Command 'python3' should match the case of the command majority (uppercase) (line 51)

 1 warning found (use --debug to expand):
 1 warning found (use --debug to expand):
 - ConsistentInstructionCasing: Command 'python3' should match the case of the command majority (uppercase) (line 51)
--------------------
--------------------
  50 |                for config_file in ../src/bootstrap/defaults/*.toml; do python3 ../x.py build --dry-run --config "$config_file"; done
  51 | >>>            python3 ../x.py check --target=i686-pc-windows-gnu --host=i686-pc-windows-gnu && \
  52 | >>>            python3 ../x.py clippy bootstrap -Dwarnings && \
  53 | >>>            python3 ../x.py clippy compiler library -Aclippy::all -Dclippy::correctness && \
  54 | >>>            python3 ../x.py build --stage 0 src/tools/build-manifest && \
  55 | >>>            python3 ../x.py test --stage 0 src/tools/compiletest && \
  56 | >>>            python3 ../x.py test --stage 0 core alloc std test proc_macro && \
  57 | >>>            # Build both public and internal documentation.
  58 | >>>            RUSTDOCFLAGS=\"--document-private-items --document-hidden-items\" python3 ../x.py doc --stage 0 library && \
  59 | >>>            mkdir -p /checkout/obj/staging/doc && \
  60 | >>>            cp -r build/x86_64-unknown-linux-gnu/doc /checkout/obj/staging && \
  61 | >>>            RUSTDOCFLAGS=\"--document-private-items --document-hidden-items\" python3 ../x.py doc --stage 0 compiler && \
  62 | >>>            RUSTDOCFLAGS=\"--document-private-items --document-hidden-items\" python3 ../x.py doc --stage 0 library/test && \
  63 | >>>            /scripts/validate-toolstate.sh && \
  64 | >>>            /scripts/validate-error-codes.sh && \
  65 | >>>            reuse --include-submodules lint && \
  66 | >>>            # Runs checks to ensure that there are no issues in our JS code.
  67 | >>>            es-check es2019 ../src/librustdoc/html/static/js/*.js && \
  68 | >>>            eslint -c ../src/librustdoc/html/static/.eslintrc.js ../src/librustdoc/html/static/js/*.js && \
  69 | >>>            eslint -c ../src/tools/rustdoc-js/.eslintrc.js ../src/tools/rustdoc-js/tester.js && \
  70 | >>>            eslint -c ../src/tools/rustdoc-gui/.eslintrc.js ../src/tools/rustdoc-gui/tester.js
--------------------
ERROR: failed to solve: dockerfile parse error on line 51: unknown instruction: python3
Command failed. Attempt 5/5:
#0 building with "elastic_dubinsky" instance using docker-container driver
#0 building with "elastic_dubinsky" instance using docker-container driver

#1 [internal] load build definition from Dockerfile
#1 transferring dockerfile: 3.37kB done
#1 WARN: ConsistentInstructionCasing: Command 'python3' should match the case of the command majority (uppercase) (line 51)

 1 warning found (use --debug to expand):
 1 warning found (use --debug to expand):
 - ConsistentInstructionCasing: Command 'python3' should match the case of the command majority (uppercase) (line 51)
--------------------
--------------------
  50 |                for config_file in ../src/bootstrap/defaults/*.toml; do python3 ../x.py build --dry-run --config "$config_file"; done
  51 | >>>            python3 ../x.py check --target=i686-pc-windows-gnu --host=i686-pc-windows-gnu && \
  52 | >>>            python3 ../x.py clippy bootstrap -Dwarnings && \
  53 | >>>            python3 ../x.py clippy compiler library -Aclippy::all -Dclippy::correctness && \
  54 | >>>            python3 ../x.py build --stage 0 src/tools/build-manifest && \
  55 | >>>            python3 ../x.py test --stage 0 src/tools/compiletest && \
  56 | >>>            python3 ../x.py test --stage 0 core alloc std test proc_macro && \
  57 | >>>            # Build both public and internal documentation.
  58 | >>>            RUSTDOCFLAGS=\"--document-private-items --document-hidden-items\" python3 ../x.py doc --stage 0 library && \
  59 | >>>            mkdir -p /checkout/obj/staging/doc && \
  60 | >>>            cp -r build/x86_64-unknown-linux-gnu/doc /checkout/obj/staging && \
  61 | >>>            RUSTDOCFLAGS=\"--document-private-items --document-hidden-items\" python3 ../x.py doc --stage 0 compiler && \
  62 | >>>            RUSTDOCFLAGS=\"--document-private-items --document-hidden-items\" python3 ../x.py doc --stage 0 library/test && \
  63 | >>>            /scripts/validate-toolstate.sh && \
  64 | >>>            /scripts/validate-error-codes.sh && \
  65 | >>>            reuse --include-submodules lint && \
  66 | >>>            # Runs checks to ensure that there are no issues in our JS code.
  67 | >>>            es-check es2019 ../src/librustdoc/html/static/js/*.js && \
  68 | >>>            eslint -c ../src/librustdoc/html/static/.eslintrc.js ../src/librustdoc/html/static/js/*.js && \
  69 | >>>            eslint -c ../src/tools/rustdoc-js/.eslintrc.js ../src/tools/rustdoc-js/tester.js && \
  70 | >>>            eslint -c ../src/tools/rustdoc-gui/.eslintrc.js ../src/tools/rustdoc-gui/tester.js
--------------------
ERROR: failed to solve: dockerfile parse error on line 51: unknown instruction: python3
The command has failed after 5 attempts.
##[error]Process completed with exit code 1.
```

</details>

--- a/src/bootstrap/defaults/config.tools.toml
+++ b/src/bootstrap/defaults/config.tools.toml
@@ -1,10 +1,6 @@
 # These defaults are meant for contributors to tools which build on the
 # compiler, but do not modify it directly.
 [rust]
-# This enables `RUSTC_LOG=debug`, avoiding confusing situations
-# where adding `debug!()` appears to do nothing.
-# However, it makes running the compiler slightly slower.
-debug-logging = true
 # This greatly increases the speed of rebuilds, especially when there are only minor changes. However, it makes the initial build slightly slower.
 incremental = true
 # Download rustc from CI instead of building it from source.

--- a/src/bootstrap/src/utils/change_tracker.rs
+++ b/src/bootstrap/src/utils/change_tracker.rs
@@ -200,4 +200,9 @@ pub const CONFIG_CHANGE_HISTORY: &[ChangeInfo] = &[
         severity: ChangeSeverity::Warning,
         summary: "`llvm.lld` is enabled by default for the dist profile. If set to false, `lld` will not be included in the dist build.",
     },
+    ChangeInfo {
+        change_id: 127913,
+        severity: ChangeSeverity::Warning,
+        summary: "`debug-logging` option has been removed from the default `tools` profile.",
+    },
 ];

--- a/src/ci/docker/host-x86_64/mingw-check/Dockerfile
+++ b/src/ci/docker/host-x86_64/mingw-check/Dockerfile
@@ -37,6 +37,7 @@ RUN sh /scripts/sccache.sh
 COPY host-x86_64/mingw-check/reuse-requirements.txt /tmp/
 RUN pip3 install --no-deps --no-cache-dir --require-hashes -r /tmp/reuse-requirements.txt
 
+COPY host-x86_64/mingw-check/check-default-config-profiles.sh /scripts/
 COPY host-x86_64/mingw-check/validate-toolstate.sh /scripts/
 COPY host-x86_64/mingw-check/validate-error-codes.sh /scripts/
 
@@ -46,8 +47,7 @@ ENV RUN_CHECK_WITH_PARALLEL_QUERIES 1
 # We disable optimized compiler built-ins because that requires a C toolchain for the target.
 # We also skip the x86_64-unknown-linux-gnu target as it is well-tested by other jobs.
 ENV SCRIPT python3 ../x.py check --stage 0 --set build.optimized-compiler-builtins=false core alloc std --target=aarch64-unknown-linux-gnu,i686-pc-windows-msvc,i686-unknown-linux-gnu,x86_64-apple-darwin,x86_64-pc-windows-gnu,x86_64-pc-windows-msvc && \
-           # Ensure that default config profiles are not broken
-           for config_file in ../src/bootstrap/defaults/*.toml; do python3 ../x.py build --dry-run --config \"$$config_file\"; done && \
+           /scripts/check-default-config-profiles.sh && \
            python3 ../x.py check --target=i686-pc-windows-gnu --host=i686-pc-windows-gnu && \
            python3 ../x.py clippy bootstrap -Dwarnings && \
            python3 ../x.py clippy compiler library -Aclippy::all -Dclippy::correctness && \

--- a/src/ci/docker/host-x86_64/mingw-check/Dockerfile
+++ b/src/ci/docker/host-x86_64/mingw-check/Dockerfile
@@ -47,7 +47,7 @@ ENV RUN_CHECK_WITH_PARALLEL_QUERIES 1
 # We also skip the x86_64-unknown-linux-gnu target as it is well-tested by other jobs.
 ENV SCRIPT python3 ../x.py check --stage 0 --set build.optimized-compiler-builtins=false core alloc std --target=aarch64-unknown-linux-gnu,i686-pc-windows-msvc,i686-unknown-linux-gnu,x86_64-apple-darwin,x86_64-pc-windows-gnu,x86_64-pc-windows-msvc && \
            # Ensure that default config profiles are not broken
-           for config_file in ../src/bootstrap/defaults/*.toml; do python3 ../x.py build --dry-run --config "$config_file"; done
+           for config_file in ../src/bootstrap/defaults/*.toml; do python3 ../x.py build --dry-run --config \"$$config_file\"; done && \
            python3 ../x.py check --target=i686-pc-windows-gnu --host=i686-pc-windows-gnu && \
            python3 ../x.py clippy bootstrap -Dwarnings && \
            python3 ../x.py clippy compiler library -Aclippy::all -Dclippy::correctness && \

--- a/src/ci/docker/host-x86_64/mingw-check/check-default-config-profiles.sh
+++ b/src/ci/docker/host-x86_64/mingw-check/check-default-config-profiles.sh
@@ -1,0 +1,12 @@
+#!/bin/bash
+# Runs bootstrap (in dry-run mode) with each default config profile to ensure they are not broken.
+
+set -euo pipefail
+
+config_dir="../src/bootstrap/defaults"
+
+# Loop through each configuration file in the directory
+for config_file in "$config_dir"/*.toml;
+do
+    python3 ../x.py check --config $config_file --dry-run
+done

